src/components/Pagination/Pagination.jsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  PaginationWrap,
+  PaginationButtonsWrap,
+  PaginationItem,
+  PaginationDots,
+  PaginationButton,
+  PaginationNavigation,
+} from './styles';
+import PrevArrow24 from '../Icons/PrevArrow24';
+import NextArrow24 from '../Icons/NextArrow24';
+
+const Pagination = ({ currentPage, total, onPageChange }) => {
+  const [buttons, setButtons] = useState([]);
+  const numberOfPages = [];
+
+  for (let i = 1; i <= total; i++) { //eslint-disable-line
+    numberOfPages.push(i);
+  }
+
+  useEffect(() => {
+    let paginationArray;
+    const leftDots = '...';
+    const rightDots = ' ...';
+
+    if (total <= 9) {
+      paginationArray = numberOfPages;
+    } else if (currentPage < 5) {
+      const sliced1 = numberOfPages.slice(0, 6);
+      const sliced2 = numberOfPages.slice(-3);
+      paginationArray = [...sliced1, leftDots, ...sliced2];
+    } else if (currentPage >= 5) {
+      const sliced1 = numberOfPages.slice(0, 3);
+      const sliced2 = numberOfPages.slice(-3);
+      const sliced3 = numberOfPages.slice(`${currentPage - 3}`, `${currentPage + 2}`);
+      if (
+        sliced3[0] - sliced1[sliced1.length - 1] > 1 &&
+        sliced2[0] - sliced3[sliced3.length - 1] > 1
+      ) {
+        paginationArray = [...sliced1, leftDots, ...sliced3, rightDots, ...sliced2];
+      } else if (sliced1[sliced1.length - 1] - sliced3[0] > 1) {
+        paginationArray = [...sliced1, leftDots, ...sliced3, ...sliced2];
+      } else if (sliced2[0] - sliced3[sliced3.length - 1] > 1) {
+        paginationArray = [...sliced1, ...sliced3, rightDots, ...sliced2];
+      } else {
+        paginationArray = [...sliced1, leftDots, ...sliced3, ...sliced2];
+      }
+    }
+
+    paginationArray = Array.from(new Set(paginationArray));
+    setButtons(paginationArray);
```

Comment:
Why do you need state here? You can create a function inside of this component that will do all stuff and return resulted array then call this function, get resulted array and draw pages.